pkg/monitoring/endpoint.go

```diff
@@ -0,0 +1,389 @@
+package monitoring
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/telekom/das-schiff-network-operator/pkg/healthcheck"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	all          = "all"
+	protocolIP   = "ip"
+	protocolIPv4 = "ipv4"
+	protocolIPv6 = "ipv6"
+
+	StatusSvcNameEnv      = "STATUS_SVC_NAME"
+	StatusSvcNamespaceEnv = "STATUS_SVC_NAMESPACE"
+)
+
+//go:generate mockgen -destination ./mock/mock_endpoint.go . FRRClient
+type FRRClient interface {
+	ExecuteWithJSON(args []string) []byte
+}
+
+type Endpoint struct {
+	cli FRRClient
+	c   client.Client
+
+	statusSvcName      string
+	statusSvcNamespace string
+}
+
+// NewEndpoint creates new endpoint object.
+func NewEndpoint(k8sClient client.Client, frrcli FRRClient, svcName, svcNamespace string) *Endpoint {
+	return &Endpoint{cli: frrcli, c: k8sClient, statusSvcName: svcName, statusSvcNamespace: svcNamespace}
+}
+
+// SetHandlers configures HTTP handlers.
+func (e *Endpoint) SetHandlers() {
+	http.HandleFunc("/show/route", e.ShowRoute)
+	http.HandleFunc("/show/bgp", e.ShowBGP)
+	http.HandleFunc("/show/evpn", e.ShowEVPN)
+	http.HandleFunc("/all/show/route", e.PassRequest)
+	http.HandleFunc("/all/show/bgp", e.PassRequest)
+	http.HandleFunc("/all/show/evpn", e.PassRequest)
```

Comment:
Can we please create our own http server instance? I saw that it's also using the default instance in main.go for the frr-exporter, please use a custom instance for everything.

Reply:
I've created separate server custom instances for both monitoring endpoint and frr-exporter.
However, adding another instance would require one of those to be run in separate goroutine which would require some sort of communication between those two (for example, if one server dies, the other one should be stopped gracefully and the pod/container should be restarted, etc). I think this is not really worth it as it makes this unnecessary complex, so I've created new `cmd/monitoring` directory and added new `monitoring` image that is run in separate container in the network-operator pod.
Do you think this will be OK, or should we keep frr-exporter and monitoring in the same image/container?

Reply:
Ugh, I'm sorry, I should've been clearer here. I didn't mean to split it into two http servers, I just wanted to not use the default http server. I even thought about it might being a bit to unclear but then decided it's good enough...
Can you change it back to a single server instance? Just don't use the default mux of the `http` package.

Reply:
No problem. :) In fact, that's my bad. I've should have thought that by 'a custom instance' (not instance**s** 😅) you've meant just one.
Anyway, it should be fixed now. :)